index.bs

```diff
@@ -3076,6 +3091,57 @@ and |session|:
 
 </div>
 
+<div algorithm="locate nodes using accessibility attributes">
+To <dfn>locate nodes using accessibility attributes</dfn> with given |context nodes|, |selector|,
+|maximum returned node count|, and |session|:
+
+1. If |selector|'s <code>role</code> and |selector|'s <code>name</code> are missing, return [=error=] with [=error code=]
+   [=invalid selector=].
+
+1. Let |returned nodes| be an empty [=/list=].
+
+1. For each |context node| in |context nodes|:
+
+  1. Let |match| be true.
+
+  1. If |selector|'s <code>role</code> is present:
```

Comment:
Not sure I understand that logic here. Shouldn't we first run some kind of selector to find specific nodes below the start nodes? Note that `context nodes` are these start nodes. And right now we would just return these which have the wanted role and name.

So shouldn't it be similar to the CSS selector but by using the `[aria-label=%label%]`.

Reply:
I believe iteration happens here:

```
  1. Let |child nodes| be an empty [=/list=] and, for each node |child| in the
     <a spec=dom>children</a> of |context node|:

    1. If |child| implements {{Element}}, [=list/append=] |child| to |child nodes|.
```

so this algorithm returns all nodes that match the predicate (role OR name matches). aria-label is not a computed attribute and, therefore, does not query in the same way.

Reply:
You are right. Interesting that I missed that.

Beside that one question. For the `locate nodes using inner text` locator we also have a `maxDepth` optional argument to limit the search on the DOM. I wonder if we should have it here as well given that computing the role and label might be expensive as well, or is that not the case?

Reply:
I am not sure: for now it was not an issue in Puppeteer but we could add it later if it becomes a problem (usually those attributes are cached for a11y purposes anyway). There is `maximum returned node count` to limit the number of matches returned and abort early and that helps with the expensiveness of the search.

Reply:
Rewrote this part.